listenbrainz/webserver/rabbitmq_connection.py

```diff
@@ -1,25 +1,39 @@
 import sys
 from time import sleep
-import pika
+from flask import Flask
+from flask.ext.pika import Pika as FPika
 
 _rabbitmq = None
 
 def init_rabbitmq_connection(app):
     """Create a connection to the RabbitMQ server."""
     global _rabbitmq
 
-    app.logger.info("Call to init_rabbit_mq_connection")
     if "RABBITMQ_HOST" not in app.config:
         app.logger.error("RabbitMQ host:port not defined. Sleeping 2 seconds, and exiting.")
         sleep(2)
         sys.exit(-1)
 
+    FLASK_PIKA_PARAMS = {
+        'host':app.config['RABBITMQ_HOST'],
+        'port': app.config['RABBITMQ_PORT'],
+        'username': 'guest',
+        'password': 'guest',
+    }
+
+    FLASK_PIKA_POOL_PARAMS = {
+        'pool_size': 8,
+        'pool_recycle': 600
+    }
+
+    app.config['FLASK_PIKA_PARAMS'] = FLASK_PIKA_PARAMS
+    app.config['FLASK_PIKA_POOL_PARAMS'] = FLASK_PIKA_POOL_PARAMS
+
     while True:
         try:
-            _rabbitmq = pika.BlockingConnection(pika.ConnectionParameters(host=app.config['RABBITMQ_HOST'], port=app.config['RABBITMQ_PORT']))
-            app.logger.info("Connected connect to rabbitmq!")
+            _rabbitmq = FPika(app)
             return
-        except pika.exceptions.ConnectionClosed:
+        except Exception as err:
```

Comment:
Why the broadening of the exception? Is there not a generic flask-pika exception that we can catch here?